include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -891,6 +896,49 @@ inline std::string CustomOpApi::KernelInfoGetAttribute<std::string>(_In_ const O
   return out;
 }
 
+template <>
+inline std::vector<float> CustomOpApi::KernelInfoGetAttributeArray(_In_ const OrtKernelInfo* info, _In_ const char* name) {
+  size_t size = 0;
+  std::vector<float> out;
+  std::string error_string_required("Result buffer is not large enough");
+
+  OrtStatus* status = api_.KernelInfoGetAttributeArray_float(info, name, nullptr, &size);
+
```

Comment:
I somewhat understand the need to call once to size a buffer for a single std::string, but would it be better to have a method to ascertain the size of the array given that's not type specific?

Reply:
Fixed so as to use the logic we discussed offline